internal/pkg/agent/application/monitoring/v1_monitor.go

```diff
@@ -772,74 +776,6 @@ func (b *BeatsMonitor) injectMetricsInput(
 			})
 		}
 
-		streams = append(streams, map[string]interface{}{
```

Comment:
I don't think you want to delete this unconditionally, instead you want a single instance of it scraping the /stats endpoint of the elastic agent process.